pkg/kando/location_delete.go

```diff
@@ -32,43 +29,20 @@ func newLocationDeleteCommand() *cobra.Command {
 		Short: "Delete artifacts from s3-compliant object storage",
 		// TODO: Example invocations
 		RunE: func(c *cobra.Command, args []string) error {
-			return runLocationDelete(c)
+			if err := validateCommandArgs(c); err != nil {
+				return err
+			}
+			dataMover, err := dataMoverFromCMD(c, kopiaSnapshotFlagName)
+			if err != nil {
+				return err
+			}
+			return dataMover.Delete(context.Background(), pathFlag(c))
```

Comment:
`datamover` can be nil here.

Reply:
We should handle not returning `nil` only from https://github.com/kanisterio/kanister/pull/1951/files#diff-f90618f09f962ed44a47af2d9876cf0c504ac5242900037f750bebf7c25632d1R69

Reply:
`datamover` could only be nil in case error is not nil and we're already handling it !